exec/csnex/Support.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace csnex
+{
+    public class Support
+    {
+        static public List<string> Paths;
+
+        public Support(string source_path)
+        {
+            Paths = new List<string>();
+            // Always check current directory;
+            Paths.Add(string.Format(".{0}", Path.DirectorySeparatorChar));
+            // Also check the Neon executable path.
+            Paths.Add(string.Format("{0}{1}", source_path, Path.DirectorySeparatorChar));
+            // Then check the cwd.
+            Paths.Add(Environment.CurrentDirectory);
+
+            // Get paths from %NEONPATH% environment variable, if there is one.
+            string ne = Environment.GetEnvironmentVariable("NEONPATH");
+            if (ne != null) {
+                string[] evpaths = ne.Split(Path.PathSeparator);
+                foreach (string s in evpaths) {
+                    // Don't bother pushing empty paths.
+                    if (s != null && s.Length > 0) {
+                        Paths.Add(s);
+                    }
+                }
+            }
+            // Next, check for .neonpath, and process any paths that might exist in it.
+            if (File.Exists(".neonpath")) {
+                StreamReader sr = new StreamReader(".neonpath");
+                while (!sr.EndOfStream) {
```

Comment:
This kind of "end of file" checking is also not robust. The `ReadLine` function already tells you when you are at the end of the file (it returns `null`). Use that instead.

Reply:
Removed sr.EndOfStream check.